lib/lte_link_control/lte_lc.c

```diff
@@ -245,7 +245,9 @@ static void at_handler_cereg(const char *response)
 
 	err = lte_lc_psm_get(&psm_cfg.tau, &psm_cfg.active_time);
 	if (err) {
-		LOG_ERR("Failed to get PSM information");
+		if (err != -EBADMSG) {
+			LOG_ERR("Failed to get PSM information");
```

Comment:
Just a suggestion, but there could be a comment that -EBADMSG is received in case the modem is no longer registered and we don't want to print an error in that case.

Reply:
Agree. I forgot to mention, the API docs for lte_lc_psm_get could also be updated for the EBADMSG return value to include this.

Reply:
Actually, it's already in the API:
```
/** @brief Function for getting the current PSM (Power Saving Mode)
 *	   configurations for periodic TAU (Tracking Area Update) and
 *	   active time, both in units of seconds.
 *
 * @param tau Pointer to the variable for parsed periodic TAU interval in
 *	      seconds. Positive integer, or -1 if timer is deactivated.
 * @param active_time Pointer to the variable for parsed active time in seconds.
 *	              Positive integer, or -1 if timer is deactivated.
 *
 * @retval 0 if successful.
 * @retval -EINVAL if input argument was invalid.
 * @retval -EFAULT if AT command failed.
 * @retval -EBADMSG if no active time and/or TAU value was received.
 */
int lte_lc_psm_get(int *tau, int *active_time);
```
For some reason the implementation has not been aligned with the API.

Reply:
Thanks guys for the comments. I was thinking that current statement in a doc is including also that case: "-EBADMSG if no active time and/or TAU value was received"
But yeah, you're right, maybe good to add clarification for this case as well. But added that only into "docs". Please check + @peknis .

Reply:
Yes, but a more descriptive message informing the user why the psm timers was not received. For instance, when not registered to the LTE network.

Reply:
> For some reason the implementation has not been aligned with the API.
Didn't notice this comment before my comment above. 
Yep, usage of that cause value -EBADMSG was removed due to some reason at some of the earlier commits.

> Yes, but a more descriptive message informing the user why the psm timers was not received. For instance, when not registered to the LTE network.
-EBADMSG is now returned always when "no active time and TAU values were received", that is case when modem does not response with those parameters, and yeah, one of the such situations is happening when modem is not registered. Previously, EFAULT was responded and also error logging in case of "normal" situation where we might be as in referenced Jira item.

Anyways, EBADMSG doc now changed to include not registered case for clarity for lte_lc_psm_get().

Reply:
Good stuff